R/await.R

```diff
@@ -109,29 +109,37 @@ await <- function(f, ...,
 }
 
 #' @param .x a vector of values to be passed to \code{f}
+#' @param .y a vector of values to be passed to \code{f} (default \code{NULL})
 #' @export
 #' @describeIn await Call a function repeatedly for all values of a vector until all have reached a completed status
-await_all <- function(f, .x, ...,
+await_all <- function(f, .x, .y = NULL, ...,
                       .status_key = "state",
                       .success_states = c("succeeded", "success"),
                       .error_states = c("failed", "cancelled"),
                       .timeout = NULL, .interval = NULL,
                       .verbose = FALSE) {
 
+
   responses <- vector(mode = "list", length = length(.x))
   called <- rep(FALSE, length(.x))
   i <- 1
   start <- Sys.time()
   fname <- as.character(substitute(f))
 
+  if (!is.null(.y) & (length(.x) != length(.y))) {
+    error <- c("Lengths of input parameters (.x and .y) are not equal!")
+    stop(error)
+  }
+
+  zipped_parameters <- if (is.null(.y)) .x else lapply(strsplit(paste(.x, .y), ' '), as.integer)
```

Comment:
I think we want `mapply(c, .x, .y, SIMPLIFY = FALSE)` here instead of `lapply(strsplit(paste(.x, .y), ' '), as.integer)`